lib/inferno/dsl/fhir_evaluation/rules/valuesets_demonstrate.rb

```diff
@@ -0,0 +1,207 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'json'
+require 'uri'
+require 'base64'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        # This rule evaluates if an IG defines a new valueset, the examples should
+        # demonstrate reasonable coverage of that valueset.
+        # Note this probably only makes sense for small valuesets
+        # such as status options, not something like disease codes from SNOMED.
+
+        # Algorithm:
+        # 1. Extract pairs of system and code from include in value sets in IG
+        # 2. If valueSet exists in include, retrieve the value sets from UMLS.
+        #    Extract pairs of system and code from the result.
+        # 3. For each pair of system and code, check if any resources in the IG have instance of them.
+        # 4. Count total number of existences.
+
+        class ValueSetsDemonstrate < Rule
+          attr_accessor :config
+
+          def check(context)
+            @config = context.config
+
+            value_set_unevaluated = []
+            value_set_used = []
+            value_set_unused = []
+
+            context.ig.resources_by_type['ValueSet'].each do |valueset|
```

Comment:
Since `ig.value_sets` now exists, may as well use it

Reply:
I believe what `ig.value_set` does is to collect `StructureDefinition` where `type` is `ValueSet` whereas the code above is to collect `resourceType` is `ValueSet` (example: https://www.hl7.org/fhir/us/core/STU3.1.1/ValueSet-birthsex.json.html) And what we want to do is the latter, is that right?

Reply:
Hmm ok yeah I misread what `ig.value_sets` is doing. It doesn't really make sense to get StructureDefinitions where type=ValueSet. Also from a quick search here I don't see anything else actually calling it, so rather than refactor more things I'd say just remove it

Reply:
Make sense. Addressed.

https://github.com/inferno-framework/inferno-core/pull/636/commits/7670a5b95459dad4be141e1e1a9a4f2e7d1a6ce8

Once this is merged, I will have to go around already merged PRs to see other cases of `value_sets` since this change of `ig` will affect them.